k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -716,11 +717,11 @@ private MessageBuilder createMessageBuilder(boolean isDraft) {
                 return null;
             }
 
-            PgpMessageBuilder pgpBuilder = new PgpMessageBuilder(getApplicationContext());
+            PgpMessageBuilder pgpBuilder = new PgpMessageBuilder(getApplicationContext(), UUIDGenerator.getInstance());
```

Comment:
The pattern we use most of the time is to have a static factory method (without arguments) that fetches or creates the dependencies and injects them as constructor arguments.